python-package/securexgboost/core.py

```diff
@@ -367,17 +397,19 @@ class DMatrix(object):
     _feature_names = None  # for previous version's pickle
     _feature_types = None
 
-    def __init__(self, data, encrypted=False, label=None, missing=None,
+    def __init__(self, data_dict, encrypted=True, label=None, missing=None,
                  weight=None, silent=False,
```

Comment:
Should we remove the parameters we don't support (or that we don't know if we support)? e.g. label, missing, weight, feature_names, feature_types, nthread

Reply:
Hmm, I think we can support these parameters. All they need is support for `get_float_info` and `set_float_info` both of which are supported by the C API. However, I've currently disabled these functions in core.py (marked with a TODO) because `get_float_info` should return encrypted values, and `set_float_info` should similarly accept encrypted values. The encryption support is pending.
Do you think we should remove the parameters until we add this support?

Reply:
Removed these args, left a TODO.